delivery_sending/models/sending_request.py

```diff
@@ -0,0 +1,147 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+# Copyright 2022 Impulso Diagonal - Javier Colmeiro
+# Copyright 2022 Tecnativa - Víctor Martínez
+import logging
+
+from suds.client import Client
+from suds.sax.text import Raw
+
+from odoo import _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class SendingRequest(object):
+    """Interface between Sending SOAP API and Odoo recordset
+       Abstract Sending API Operations to connect them with Odoo
+    """
+
+    def __init__(self, carrier):
```

Comment:
Non blocking, but you know I like to abstract the request class away from any odoo object :grin:

Reply:
Yes, I know, but I think just the opposite.

Reply:
I'm with @chienandalu. This class is not ORM (`models.Model`), but Python one (`object`), so only Python arguments should be passed for eventually extracting this isolatedly from Odoo.